cmd/tink-cli/cmd/hardware/all.go

```diff
@@ -24,7 +25,11 @@ var allCmd = &cobra.Command{
 		var hw *hardware.Hardware
 		err = nil
 		for hw, err = alls.Recv(); err == nil && hw != nil; hw, err = alls.Recv() {
-			fmt.Println(hw.JSON)
+			b, err := json.Marshal(hw)
+			if err != nil {
+				log.Fatal(err)
```

Comment:
This is just to retrieve hardware info, may be one hardware data cannot be converted to json properly, we can just log an error and go on for next. Why do we have to log fatal, which will crash the cli agent and make it harder to debug.